src/core/profiles/service/language.action.ts

```diff
@@ -0,0 +1,87 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { Language } from 'src/core/language/entities/language.entity';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { LanguageProfile } from '../entities/language-profile.entity';
+import { Profile } from '../entities/profile.entity';
+import { AddLanguageDto } from '../dto/add-language.dto';
+import { UserActiveInterface } from 'src/common/interface/user-active-interface';
+import {
+  ERROR_EXIST_LANGUAGE_IN_PROFILE,
+  ERROR_NOT_FOUND_LANGUAGE_IN_PROFILE,
+  ERROR_PROFILE_LANGUAGUE_NOT_FOUND,
+  PROFILE_NOT_FOUND,
+} from '../messages';
+
+@Injectable()
+export default class LanguageAction {
+  constructor(
+    @InjectRepository(Language)
+    private readonly languageRepository: Repository<Language>,
+
+    @InjectRepository(LanguageProfile)
+    private readonly languageProfileRepository: Repository<LanguageProfile>,
+
+    @InjectRepository(Profile)
+    private readonly profileRepository: Repository<Profile>,
+  ) {}
+
+  async add(addLanguageProfile: AddLanguageDto, user: UserActiveInterface) {
```

Comment:
Maybe, para evitar mal entendidos
```suggestion
  async add(addLanguageProfile: AddLanguageProfileDto, user: UserActiveInterface) {
```